usr/src/uts/common/fs/zfs/zap_micro.c

```diff
@@ -527,9 +526,9 @@ zap_lockdir(objset_t *os, uint64_t obj, dmu_tx_t *tx,
 			dprintf("upgrading obj %llu: num_entries=%u\n",
 			    obj, zap->zap_m.zap_num_entries);
 			*zapp = zap;
-			return (mzap_upgrade(zapp, tx, 0));
+			return (mzap_upgrade(zapp, tag, tx, 0));
 		}
-		err = dmu_object_set_blocksize(os, obj, newsz, 0, tx);
+		int err = dmu_object_set_blocksize(os, obj, newsz, 0, tx);
```

Comment:
This change causes an unused variable warning on `err` in ZoL for non-debug builds since ASSERTs are disabled.